ddtrace/contrib/asyncpg/patch.py

```diff
@@ -0,0 +1,151 @@
+from typing import TYPE_CHECKING
+
+from ddtrace import Pin
+from ddtrace import config
+from ddtrace.vendor import wrapt
+
+from ...constants import SPAN_MEASURED_KEY
+from ...ext import SpanTypes
+from ...ext import db
+from ...ext import net
+from ...internal.logger import get_logger
+from ...internal.utils import get_argument_value
+from ..trace_utils import ext_service
+from ..trace_utils import unwrap
+from ..trace_utils import with_traced_module as with_traced_module_sync
+from ..trace_utils import wrap
+from ..trace_utils_async import with_traced_module
+
+
+if TYPE_CHECKING:
+    from types import ModuleType
+    from typing import Dict
+
+    import asyncpg
+
+config._add(
+    "asyncpg",
+    dict(
+        _default_service="postgres",
+    ),
+)
+
+
+log = get_logger(__name__)
+
+
+def _get_connection_tags(conn):
+    # type: (asyncpg.Connection) -> Dict[str, str]
+    addr = conn._addr
+    params = conn._params
+    host = port = ""
+    if isinstance(addr, tuple) and len(addr) == 2:
+        host, port = addr
+    return {
+        net.TARGET_HOST: host,
+        net.TARGET_PORT: port,
+        db.USER: params.user,
+        db.NAME: params.database,
+    }
+
+
+class _TracedConnection(wrapt.ObjectProxy):
```

Comment:
basically just async dbapi 2 ?

Reply:
not quite, the api is quite different. Connection objects in dbapi don't really do any queries, all queries are done via the cursor